addons/binding/org.openhab.binding.lgwebos/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="lgwebos" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="WebOSTV">
+		<label>WebOS TV</label>
+		<description>WebOS based smart TV</description>
+
+		<channels>
+			<channel id="power" typeId="commandType">
+				<label>Power</label>
+				<description>Via this binding TV can only be powered off, not on.</description>
+			</channel>
+			<channel id="mute" typeId="commandType">
+				<label>Mute</label>
+				<description>Current Mute Setting</description>
+			</channel>
+			<channel id="volume" typeId="volumeChannelType" />
+			<channel id="channel" typeId="stringType">
+				<label>Channel Number</label>
+				<description>Current Channel Setting</description>
+			</channel>
+			<channel id="channelUp" typeId="commandType">
+				<label>Channel Up</label>
+				<description>One Channel Up</description>
+			</channel>
+			<channel id="channelDown" typeId="commandType">
+				<label>Channel Down</label>
+				<description>One Channel Down</description>
+			</channel>
+			<channel id="channelName" typeId="channelNameChannelType" />
+			<channel id="toast" typeId="stringType">
+				<label>Toast</label>
+				<description>Send a message onto the TV screen.</description>
+			</channel>
+			<channel id="mediaPlayer" typeId="mediaPlayerType" />
+			<channel id="mediaStop" typeId="commandType">
+				<label>Stop</label>
+				<description>Stop Playback</description>
+			</channel>
+			<channel id="appLauncher" typeId="stringType">
+				<label>Application</label>
+				<description>Start application and monitor running applications.</description>
+			</channel>
+		</channels>
+
+		<!-- property as IP must not be changed -->
```

Comment:
IP is clearly a required configuration parameter, so it must not be a property.

Reply:
What is the difference between a property and a config parameter? 
I would think that users can modify the parameter.

Reply:
A property is used for static meta data that can be set in the thing xml or when a thing is created  when discovered. It can not be changed later (It can but is not persisted if I read it correctly in the documentation) For example a serial number. A config parameter is for device specific settings, that could change or are user specific, like polling times or something.  Users can modify this parameter. An ip address is something that could change I presume so it's logical to have is as a config parameter, otherwise when it would change the user needs to re-add the binding.

Reply:
To even clarify a bit further: Configuration parameters configure the HANDLER, not the remote device. So the configuration needs all information that is required by the handler code to access the device. The IP address is very clearly such a piece of information. (see also https://github.com/eclipse/smarthome/issues/3484 if you think this is weird.)